src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,39 @@
 package mate.academy;
 
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final CopyOnWriteArrayList<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService;
 
+    public EventManager() {
+        executorService = ForkJoinPool.commonPool();
     }
 
-    public void deregisterListener(EventListener listener) {
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
+    }
 
+    public void deregisterListener(EventListener listener) {
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        CompletableFuture.runAsync(() -> {
+            for (EventListener listener : listeners) {
+                listener.onEvent(event);
+            }
+        }, executorService).thenRun(() -> {
+            for (EventListener listener : listeners) {
+                System.out.println(listener.getProcessedEvent());
+            }
+        });
```

Comment:
There is a potential issue with the `notifyEvent` method. Although `CopyOnWriteArrayList` is used, which is thread-safe, modifications to the list (like adding or removing listeners) during the iteration could lead to unexpected behavior. Consider synchronizing the block or using a different approach to ensure that the list is not modified while being iterated over.